l10n_it_edi_extension/__init__.py

```diff
@@ -0,0 +1,971 @@
+                Markup().join(
+                    nl2br_enclose(" ".join(tag.split()), "li")
+                    for tag in [
+                        f'CondizioniPagamento: {term_code or "N/A"}',
+                        f'Beneficiario: {recipient or "N/A"}',
+                        f'ModalitaPagamento: {method_code or "N/A"}',
+                        f'DataRiferimentoTerminiPagamento: {payment_term_start or "N/A"}',  # noqa: E501
+                        f'GiorniTerminiPagamento: {payment_days or "N/A"}',
+                        f'DataScadenzaPagamento: {payment_due_date or "N/A"}',
+                        f'ImportoPagamento: {payment_amount or "N/A"}',
+                        f'CodUfficioPostale: {post_office_code or "N/A"}',
+                        f'CognomeQuietanzante: {recepit_surname or "N/A"}',
+                        f'NomeQuietanzante: {recepit_name or "N/A"}',
+                        f'CFQuietanzante: {recepit_cf or "N/A"}',
+                        f'TitoloQuietanzante: {recepit_title or "N/A"}',
+                        f'IstitutoFinanziario: {payment_bank_name or "N/A"}',
+                        f'IBAN: {payment_bank_iban or "N/A"}',
+                        f'ABI: {payment_bank_abi or "N/A"}',
+                        f'CAB: {payment_bank_cab or "N/A"}',
+                        f'BIC: {payment_bank_bic or "N/A"}',
+                        f'ScontoPagamentoAnticipato: {prepayment_discount or "N/A"}',
+                        f'DataLimitePagamentoAnticipato: {max_payment_date or "N/A"}',
+                        f'PenalitaPagamentiRitardati: {penalty_amount or "N/A"}',
+                        f'DataDecorrenzaPenale: {penalty_date or "N/A"}',
+                        f'CodicePagamento: {payment_code or "N/A"}',
+                    ]
+                )
+            )
+            message = Markup("{} {}<br/>{}").format(
+                "DatiPagamento", env._("from XML file:"), payment_tags
+            )
+            move.sudo().message_post(body=message)
+
+
+def _l10n_it_fatturapa_post_migration(env):
+    query = """
+        UPDATE res_partner
+        SET
+            l10n_it_pa_index = codice_destinatario,
+            l10n_it_pec_email = pec_destinatario
+    """
+    openupgrade.logged_query(env.cr, query)
+
+    query = """
+        UPDATE res_company
+        SET l10n_it_tax_system = fp.code
+        FROM res_partner rp
+        LEFT JOIN fatturapa_fiscal_position fp ON rp.register_fiscalpos = fp.id
+        WHERE res_company.partner_id = rp.id AND rp.register_fiscalpos IS NOT NULL
+    """
+    openupgrade.logged_query(env.cr, query)
+
+    query = """
+        UPDATE ir_attachment
+        SET res_model = 'account.move', res_id = fa.invoice_id
+        FROM fatturapa_attachments fa
+        WHERE ir_attachment.id = fa.ir_attachment_id
+    """
+    openupgrade.logged_query(env.cr, query)
+
+    env.cr.execute("SELECT * FROM fatturapa_activity_progress LIMIT 1")
```

Comment:
se lo scopo di `SELECT * FROM fatturapa_activity_progress LIMIT 1` è di verificare se ci sia almeno una riga non credo sia necessario dato che in quel caso la SELECT non ritornerebbe nessun record.
D'altra parte in un caso reale di migrazione mi ha dato problemi perché `fatturapa_activity_progress` non è detto che esista come tabella, per cui ho modificato la condizione così:

`if table_exists(env.cr, 'fatturapa_activity_progress'):`
    [...]

Stesso discorso per `fatturapa_summary_data` poco più sotto